nilearn/glm/tests/test_utils.py

```diff
@@ -31,12 +37,101 @@ def test_full_rank():
+    p_val = np.array(np.minimum(np.maximum(p_val, 1.e-300), 1. - 1.e-16))
+    cdf_val = np.array(np.minimum(np.maximum(cdf_val, 1.e-300), 1. - 1.e-16))
+    # Compute z-score from the p-value estimated with the SF
+    z_val_sf = norm.isf(p_val)
+    # Compute z-score from the p-value estimated with the CDF
+    z_val_cdf = norm.ppf(cdf_val)
+    # Create the final array of z-scores, ...
+    z_val = np.zeros(p_val.size)
+    # ... in which z-scores < 0 estimated w/ SF are replaced by z-scores < 0
+    # estimated w/ CDF
+    z_val[np.atleast_1d(z_val_sf < 0)] = z_val_cdf[z_val_sf < 0]
+    # ... and z-scores >=0 estimated from SF are kept.
+    z_val[np.atleast_1d(z_val_sf >= 0)] = z_val_sf[z_val_sf >= 0]
+    # Test 'z_score' function in 'nilearn/glm/contrasts.py'
+    assert_array_almost_equal(z_score(p_val, one_minus_pvalue=cdf_val), z_val)
+    # ##################### Check the numerical precision #####################
+    for t in [33.75, -8.3]:
+        p = sps.t.sf(t, 1e10)
+        cdf = sps.t.cdf(t, 1e10)
+        z_sf = norm.isf(p)
+        z_cdf = norm.ppf(cdf)
+        if p <= .5:
+            z = z_sf
+        else:
+            z = z_cdf
+        assert_array_almost_equal(z_score(p, one_minus_pvalue=cdf), z)
+
+
+def test_z_score_opposite_contrast():
+    fmri, mask = generate_fake_fmri(shape=(50, 20, 50), length=96)
+
+    nifti_masker = NiftiMasker(mask_img=mask)
+    data = nifti_masker.fit_transform(fmri)
+
+    frametimes = np.linspace(0, (96 - 1) * 2, 96)
+
+    for i in [0, 20]:
+        design_matrix = make_first_level_design_matrix(
+            frametimes, hrf_model='spm',
+            add_regs=np.array(data[:, i]).reshape(-1, 1))
+        c1 = np.array([1] + [0] * (design_matrix.shape[1] - 1))
+        c2 = np.array([0] + [1] + [0] * (design_matrix.shape[1] - 2))
+        contrasts = {'seed1 - seed2': c1 - c2, 'seed2 - seed1': c2 - c1}
+        fmri_glm = FirstLevelModel(t_r=2., 
+                                   noise_model='ar1', 
+                                   standardize=False, 
+                                   hrf_model='spm', 
+                                   drift_model='cosine')
+        fmri_glm.fit(fmri, design_matrices=design_matrix)
+        z_map_seed1_vs_seed2 = fmri_glm.compute_contrast(
+            contrasts['seed1 - seed2'], output_type='z_score')
+        z_map_seed2_vs_seed1 = fmri_glm.compute_contrast(
+            contrasts['seed2 - seed1'], output_type='z_score')
+
+        print(z_map_seed1_vs_seed2.get_data().min())
+        print(-z_map_seed2_vs_seed1.get_data().max())
+        npt.assert_almost_equal(z_map_seed1_vs_seed2.get_data().min(),
+                                -z_map_seed2_vs_seed1.get_data().max(),
+                                decimal=10)
+        print(z_map_seed1_vs_seed2.get_data().max())
+        print(-z_map_seed2_vs_seed1.get_data().min())
+        npt.assert_almost_equal(z_map_seed1_vs_seed2.get_data().max(),
```

Comment:
```assert_almost_equal``` is already imported without a need of numpy testing import ```npt```

Reply:
Yes, indeed. I have corrected that.